changfeedccl: deflake TestAlterChangefeedTelemetry

stevendanna opened the pull request:
The job system clears the lease asyncronously after cancelation. This
lease clearing transaction can cause a restart in the alter changefeed
transaction, which will lead to different feature counter
counts. Thus, we want to wait for the lease clear. However, the lease
clear isn't guaranteed to happen, so we only wait a few seconds for
it.

Epic: None

Release note: None

stevendanna committed:
testutils: add job id to error message

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/91884)
<!-- Reviewable:end -->

stevendanna committed:
changfeedccl: deflake TestAlterChangefeedTelemetry

The job system clears the lease asyncronously after cancelation. This
lease clearing transaction can cause a restart in the alter changefeed
transaction, which will lead to different feature counter
counts. Thus, we want to wait for the lease clear. However, the lease
clear isn't guaranteed to happen, so we only wait a few seconds for
it.

Release note: None

stevendanna commented:
@miretskiy I pushed another commit to de-flake the test where I found this. Mind taking a look.

miretskiy commented on pkg/ccl/changefeedccl/alter_changefeed_test.go:
```diff
@@ -423,11 +422,34 @@ func TestAlterChangefeedTelemetry(t *testing.T) {
+		// clear. However, the lease clear isn't guaranteed to
+		// happen, so we only wait a few seconds for it.
+		waitForNoLease := func() {
+			ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
```
Should the period be increased if under race?
helpers_test.go has withTimeout function that can help with this, plus assertPayloadsTimeout() is a helper that returns timeout adjusted for race (might need renaming)
